include/odp/api/spec/schedule_types.h

```diff
@@ -159,6 +165,107 @@ typedef	struct odp_schedule_param_t {
+	uint32_t max_ordered_locks;
+
+	/** Maximum number of scheduling groups */
+	unsigned max_groups;
+
+	/** Number of scheduling priorities */
+	unsigned prios;
+
+	/** Scheduled queue capabilities */
+	struct {
+		/** Maximum number of scheduled (ODP_BLOCKING) queues of the
+		  * default size. */
+		uint32_t max_num;
+
+		/** Maximum number of events a scheduled (ODP_BLOCKING) queue
+		  * can store simultaneously. The value of zero means that
+		  * scheduled queues do not have a size limit, but a single
+		  * queue can store all available events. */
+		uint32_t max_size;
+
+		/** Lock-free (ODP_NONBLOCKING_LF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Lock-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} lockfree;
+
+		/** Wait-free (ODP_NONBLOCKING_WF) implementation capabilities.
+		  * The specification is the same as for the blocking
+		  * implementation. */
+		struct {
+			/** Maximum number of queues. Wait-free queues are not
+			  * supported when zero. */
+			uint32_t max_num;
+
+			/** Maximum queue size */
+			uint32_t max_size;
+
+		} waitfree;
+
+		/** Maximum supported flows per queue.
+		 * Specifies the maximum number of flows per queue supported by
+		 * the implementation.  A value of 0 indicates flow aware mode
+		 * is not supported.
+		 */
+		uint32_t max_flows;
+
+	} queue;
+
+} odp_schedule_capability_t;
+
+/**
+ * Schedule configuration
+ */
+typedef struct odp_schedule_config_t {
+	/** Maximum number of schedule queues to be supported.
+	 * Application configures the maximum number of scheduled queues
+	 * required by the application.
+	 * @see odp_queue_capability_t
+	 */
+	uint32_t num_queues;
+
+	/** Maximum number of events required to be stored simultaneously in
+	 * schedule queue. This number must not exceed 'capa.queue.max_size'
+	 * required by the application.
+	 * A value of 0 configures default queue size supported by the
+	 * implementation.
+	 */
+	uint32_t queue_size;
+
+	/** Number of flows per queue to be supported.
+	 * Scheduler enables flow aware mode when flow count is configured
+	 * greater than 1 (up to 'capa.queue.max_flows').
+	 *
+	 * Flows are lightweight entities and events can be assigned to
+	 * specific flows by the application using odp_event_flow_id_set()
+	 * before enqueuing the event into the scheduler. This value is
+	 * ignored unless scheduler supports flow aware mode.
+	 *
+	 * Depending on the implementation this number might be rounded up to
+	 * nearest supported value (e.g power of 2). odp_schedule_config() will
+	 * update passed config with the value actually used. This number
+	 * should be less than maximum flow supported by the implementation.
+	 *
+	 * @see odp_schedule_capability_t
+	 */
+	uint32_t num_flows;
+
+} odp_schedule_config_t;
```

Comment:
Maybe "struct { ... } queue" would make sense here also. Config would be closer to capa, and later we can add num_prio, etc and lockfree/waitfree queue params.

Reply:
ack